src/3rdparty/kmessagewidget/kmessagewidget.cpp

```diff
@@ -139,7 +139,7 @@ void KMessageWidgetPrivate::createLayout()
             // Use an additional layout in row 1 for the buttons.
             auto *buttonLayout = new QHBoxLayout;
             buttonLayout->addStretch();
-            Q_FOREACH (QToolButton *button, buttons) {
+            for (QToolButton *button : buttons) {
```

Comment:
`buttons` being non-const the container will `detach()`. This requires `qAsConst(buttons)`.